### PR TITLE
Remove console warnings in tests

### PR DESCRIPTION
**Pull request by lazytype**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Clean up console errors from tests](https://jira.mongodb.org/browse/PD-712)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

~- [x] I have added tests that prove my fix is effective or that my feature works~
~- [ ] I have added necessary documentation (if appropriate)~
~- [ ] I have run `yarn changeset` and documented my changes~

~- [ ] I have added my new package to the global tsconfig~
~- [ ] I have added my new package to the Table of Contents on the global README~

### CODE REVIEW

File: packages/box/src/Box.spec.tsx [hunk 1]
```diff
@@ -182,7 +183,8 @@ describe('packages/box', () => {
   });
 
   describe('packages/box/ExtendableBox', () => {
-    describe('the types work as expected', () => {
+    // eslint-disable-next-line jest/no-disabled-tests
+    describe.skip('the types work as expected', () => {
```
Comment (brookescarlett) [inline packages/box/src/Box.spec.tsx:187]: 👍 smart

**Comment by brookescarlett** [comment]
This might be my favorite PR to date. I just ran the test suite locally and it was 😇! To note, there are still console.error's coming from `<TextInput />` but that might be a function of this PR being out of date with master? Not sure.

File: packages/box/src/Box.spec.tsx (hunk 1 above)
Comment (hswolff) [reply packages/box/src/Box.spec.tsx:187]: Were we talking about making a more concrete way and structure for type tests? Probably need a ticket to follow up on that discretely?

File: packages/mongo-nav/tsconfig.json [hunk 2]
```diff
@@ -25,6 +25,9 @@
     {
       "path": "../icon"
     },
+    {
+      "path": "../testing-lib"
```
Comment (hswolff) [inline packages/mongo-nav/tsconfig.json:29]: stack diffs 🙈

Comment (lazytype) [reply packages/mongo-nav/tsconfig.json:29]: 😮 did I miss something? this PR introduces the mongo-nav changes, so it seems like the right place to update the mongo-nav tsconfig

File: packages/box/src/Box.spec.tsx (hunk 1 above)
Comment (lazytype) [reply packages/box/src/Box.spec.tsx:187]: 👍 https://jira.mongodb.org/browse/PD-740

File: packages/mongo-nav/tsconfig.json (hunk 2 above)
Comment (hswolff) [reply packages/mongo-nav/tsconfig.json:29]: Oh, this IS the stacked diff. My bad! Thought this was into master.

File: packages/radio-box-group/src/RadioBoxGroup.spec.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ afterAll(cleanup);
 const className = 'test-radio-box-class';
 
 const { container } = render(
-  <RadioBox value="radio-1" className={className} checked={true}>
+  <RadioBox value="radio-1" className={className} checked readOnly>
```
Comment (DesignerDave) [inline packages/radio-box-group/src/RadioBoxGroup.spec.tsx:12]: I'm curious why we're adding the read-only attribute here. Seems unrelated to the PR.

Comment (lazytype) [reply packages/radio-box-group/src/RadioBoxGroup.spec.tsx:12]: This was to silence:
>Warning: Failed form propType: You provided a checked prop to a form field without an onChange handler. This will render a read-only field. If the field should be mutable use defaultChecked. Otherwise, set either onChange or readOnly

Comment (DesignerDave) [reply packages/radio-box-group/src/RadioBoxGroup.spec.tsx:12]: Ahh nice 👍
